pom.xml

```diff
@@ -205,57 +264,34 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-compiler-plugin</artifactId>
-        <version>2.5.1</version>
         <configuration>
-          <encoding>UTF-8</encoding>
           <source>${java.target}</source>
           <target>${java.target}</target>
         </configuration>
       </plugin>
-      <!-- we need to create a uber-jar for https://github.com/ktoso/maven-git-commit-id-plugin/issues/196 -->
+
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-shade-plugin</artifactId>
-        <version>2.3</version>
+        <artifactId>maven-plugin-plugin</artifactId>
         <executions>
           <execution>
-            <phase>package</phase>
+            <id>default-descriptor</id>
+            <phase>process-classes</phase>
+          </execution>
+<!--
+          <execution>
+            <id>generated-helpmojo</id>
             <goals>
-              <goal>shade</goal>
+              <goal>helpmojo</goal>
             </goals>
             <configuration>
-              <createDependencyReducedPom>false</createDependencyReducedPom>
-              <transformers>
-                <transformer implementation="org.apache.maven.plugins.shade.resource.ManifestResourceTransformer">
-                  <mainClass>pl.project13.maven.git.GitCommitIdMojo</mainClass>
-                </transformer>
-              </transformers>
-              <artifactSet>
-                <includes>
-                  <include>org.slf4j:*</include>
-                </includes>
-              </artifactSet>
+              <goalPrefix>git-commit-id</goalPrefix>
+              <helpPackageName>pl.project13.maven</helpPackageName>
             </configuration>
           </execution>
+-->
```

Comment:
It might be a fine balance. Commented out code is a ballast (false positives for searches, for example), but can be useful once in a while to avoid searching for a solution. However, it's always a ballast, while a solution - only once in a while and that one might not be the best (since it's already commented out). That tips the scale in favor of removal.